### PR TITLE
RDCC-5182 Added Version check

### PR DESCRIPTION
**Pull request by nayeemshaik-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDCC-5182

### Change description ###
https://tools.hmcts.net/jira/browse/RDCC-5182

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by nayeemshaik-hmcts — COMMENTED** [review]
Block the Merge of PR until confirmed.

File: charts/rd-caseworker-ref-api/values.yaml [hunk 1]
```diff
@@ -23,6 +23,7 @@ java:
     OPEN_ID_API_BASE_URI: https://idam-web-public.{{ .Values.global.environment }}.platform.hmcts.net/o
     REFRESH_PAGE_SIZE: 20
     REFRESH_SORT_COLUMN: caseWorkerId
+    STAFF_DATA_FILE_VERSION: v1.14
```
Comment (kavithagade12) [inline charts/rd-caseworker-ref-api/values.yaml:26]: Could you please move the above config to flux

**Review by kavithagade12 — DISMISSED** [review]

File: charts/rd-caseworker-ref-api/values.yaml (hunk 1 above)
Comment (nayeemshaik-hmcts) [reply charts/rd-caseworker-ref-api/values.yaml:26]: This is require for the PR testing. We cant inject the value from cnp-flux

Comment (nayeemshaik-hmcts) [reply charts/rd-caseworker-ref-api/values.yaml]: @kavithagade12 Moved the default value to preview yaml file

File: src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/CaseWorkerRefFunctionalTest.java [hunk 2]
```diff
@@ -359,11 +375,11 @@ public void shouldUploadServiceRoleMappingXlsFileSuccessfully() throws IOExcepti
     @ToggleEnable(mapKey = CASEWORKER_FILE_UPLOAD, withFeature = true)
     @ExtendWith(FeatureToggleConditionExtension.class)
     @Order(3)
-    public void shouldUploadServiceRoleMappingAba1XlsFileSuccessfully() throws IOException {
+    void shouldUploadServiceRoleMappingAba1XlsFileSuccessfully() throws IOException {
         ExtractableResponse<Response> uploadCaseWorkerFileResponse =
                 uploadCaseWorkerFile("src/functionalTest/resources/ServiceRoleMapping_ABA1.xls",
                         200, IDAM_ROLE_MAPPINGS_SUCCESS, TYPE_XLS,
-                        ROLE_CWD_ADMIN);
+                        ROLE_CWD_ADMIN,Boolean.TRUE);
```
Comment (SujeethKumarAri) [inline src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/CaseWorkerRefFunctionalTest.java:382]: you can set this to false as for service to idam mapping file there is no need to update version

File: src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/CaseWorkerRefFunctionalTest.java [hunk 3]
```diff
@@ -341,11 +357,11 @@ public void shouldUploadServiceRoleMappingXlsxFileSuccessfully() throws IOExcept
     @ToggleEnable(mapKey = CASEWORKER_FILE_UPLOAD, withFeature = true)
     @ExtendWith(FeatureToggleConditionExtension.class)
     @Order(2)
-    public void shouldUploadServiceRoleMappingXlsFileSuccessfully() throws IOException {
+    void shouldUploadServiceRoleMappingXlsFileSuccessfully() throws IOException {
         ExtractableResponse<Response> uploadCaseWorkerFileResponse =
                 uploadCaseWorkerFile("src/functionalTest/resources/ServiceRoleMapping_BBA9.xls",
                         200, IDAM_ROLE_MAPPINGS_SUCCESS, TYPE_XLS,
-                        ROLE_CWD_ADMIN);
+                        ROLE_CWD_ADMIN,Boolean.TRUE);
```
Comment (SujeethKumarAri) [inline src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/CaseWorkerRefFunctionalTest.java:364]: you can set this to false as for service to idam mapping file there is no need to update version

File: src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/CaseWorkerRefFunctionalTest.java [hunk 4]
```diff
@@ -322,11 +338,11 @@ public void shouldUploadXlsFileSuccessfully() throws IOException {
     @ToggleEnable(mapKey = CASEWORKER_FILE_UPLOAD, withFeature = true)
     @ExtendWith(FeatureToggleConditionExtension.class)
     @Order(1)
-    public void shouldUploadServiceRoleMappingXlsxFileSuccessfully() throws IOException {
+    void shouldUploadServiceRoleMappingXlsxFileSuccessfully() throws IOException {
         ExtractableResponse<Response> uploadCaseWorkerFileResponse =
                 uploadCaseWorkerFile("src/functionalTest/resources/ServiceRoleMapping_BBA9.xlsx",
                         200, IDAM_ROLE_MAPPINGS_SUCCESS, TYPE_XLS,
-                        ROLE_CWD_ADMIN);
+                        ROLE_CWD_ADMIN,Boolean.TRUE);
```
Comment (SujeethKumarAri) [inline src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/CaseWorkerRefFunctionalTest.java:345]: you can set this to false as for service to idam mapping file there is no need to update version

File: charts/rd-caseworker-ref-api/values.preview.template.yaml [hunk 5]
```diff
@@ -24,6 +24,7 @@ java:
     CWRD_DATA_PER_MESSAGE: 50
     LAUNCH_DARKLY_ENV: "preview"
     ENVIRONMENT_NAME: "preview"
+    STAFF_DATA_FILE_VERSION: v1.14
```
Comment (kavithagade12) [inline charts/rd-caseworker-ref-api/values.preview.template.yaml:27]: Please move it to cnp-flux

**Comment by nayeemshaik-hmcts** [comment]
Created new pull request

**Pull request closed without merging**